pkg/controllers/managementuserlegacy/helm/common/extra_args_injection.go

```diff
@@ -35,7 +35,7 @@ func injectDefaultRegistry(obj *v3.App) map[string]string {
 		return nil
 	}
 
-	return map[string]string{"systemDefaultRegistry": reg}
+	return map[string]string{"systemDefaultRegistry": reg, "cattle.systemDefaultRegistry": reg}
```

Comment:
Note: the value will be ignored by the app which values.yaml does not define the field `global.cattle.systemDefaultRegistry`